### PR TITLE
Handle missing xfb execution mode in addCallInstForXfbOutput

### PR DESCRIPTION
**Pull request by xuechen417**
There is a case that the vertex outputs has xfb decoration but without declared with xfb execution mode. As vulkan spec, these outputs will not be captured by xfb buffer. This change will add the missing handling for this case in `addCallInstForXfbOutput`. The check for existing xfb outputis based on lgc.xfb.state metadata.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
d4d9429f681bf5b31655c1c817f9d419631a7042 Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/llpc/lower/llpcSpirvLowerGlobal.cpp:2431:5: error: ‘haseXfbOut’ was not declared in this scope; did you mean ‘hasXfbOut’?
 2431 |     haseXfbOut = false;
      |     ^~~~~~~~~~
      |     hasXfbOut
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/llpc/lower/llpcSpirvLowerGlobal.cpp:2424:8: error: unused variable ‘pipelineBuildInfo’ [-Werror=unused-variable]
 2424 |   auto pipelineBuildInfo = static_cast<const Vkgc::GraphicsPipelineBuildInfo *>(m_context->getPipelineBuildInfo());
      |        ^~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
[338/581] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/lower/llpcSpirvLowerInternalLibraryIntrinsicUtil.cpp.o
[339/581] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/lower/llpcSpirvLowerRayQueryPostInline.cpp.o

File: llpc/lower/llpcSpirvLowerGlobal.cpp [hunk 1]
```diff
@@ -2423,7 +2423,16 @@ void SpirvLowerGlobal::addCallInstForXfbOutput(const ShaderInOutMetadata &output
   assert(m_shaderStage == m_lastVertexProcessingStage);
   auto pipelineBuildInfo = static_cast<const Vkgc::GraphicsPipelineBuildInfo *>(m_context->getPipelineBuildInfo());
   DenseMap<unsigned, Vkgc::XfbOutInfo> *locXfbMapPtr = outputMeta.IsBuiltIn ? &m_builtInXfbMap : &m_genericXfbMap;
-  if (pipelineBuildInfo->apiXfbOutData.forceDisableStreamOut || (locXfbMapPtr->empty() && !outputMeta.IsXfb))
+  bool hasXfbOut = m_entryPoint->getMetadata(lgc::XfbStateMetadataName);
+#if LLPC_CLIENT_INTERFACE_MAJOR_VERSION < 69
+  // NOTE: primitive generated query will create xfb metadata in the old version. It may be a fake XFB if there is no
+  // XFB info from API interface or output metadata.
+  if (hasXfbOut && locXfbMapPtr->empty() && !outputMeta.IsXfb) {
+    haseXfbOut = false;
```
Comment (nhaehnle) [inline llpc/lower/llpcSpirvLowerGlobal.cpp:2431]: Typo: hasXfbOut

**Review by nhaehnle — COMMENTED** [review]
LGTM modulo compile error.

File: llpc/lower/llpcSpirvLowerGlobal.cpp (hunk 1 above)
Comment (xuechen417) [reply llpc/lower/llpcSpirvLowerGlobal.cpp:2431]: Thanks. Fixed.

**Comment by xuechen417** [comment]
V1: Fix a typo.

**Comment by amdvlk-admin** [comment]
806222ef138ae6d645099df6bb58e064e77e8f88 Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/llpc/lower/llpcSpirvLowerGlobal.cpp:2424:8: error: unused variable ‘pipelineBuildInfo’ [-Werror=unused-variable]
 2424 |   auto pipelineBuildInfo = static_cast<const Vkgc::GraphicsPipelineBuildInfo *>(m_context->getPipelineBuildInfo());
      |        ^~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
[94/320] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/translator/lib/SPIRV/libSPIRV/SPIRVStream.cpp.o
[95/320] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/translator/lib/SPIRV/libSPIRV/SPIRVInstruction.cpp.o

**Comment by amdvlk-admin** [comment]
### Test summary for commit c7463c1f8b12a64c3b0d657125ff3e4fbf513308
<details><summary>CTS tests (Failed: 0/138994)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu navi3x, Srdcvk</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu navi2x, Srdcvk</summary><ul>

<li>Passed: 41177/69491 (59.3%)</li>
<li>Failed: 0/69491 (0.0%)</li>
<li>Not Supported: 28314/69491 (40.7%)</li><li>Warnings: 0/69491 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit cd520632870beff47e8b7a84bb9fa21d89424367
<details><summary>CTS tests (Failed: 0/138995)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu navi3x, Srdcvk</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu navi2x, Srdcvk</summary><ul>

<li>Passed: 41178/69491 (59.3%)</li>
<li>Failed: 0/69491 (0.0%)</li>
<li>Not Supported: 28313/69491 (40.7%)</li><li>Warnings: 0/69491 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by xuechen417** [comment]
V2: refine the logic of checking xfb existing.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 372d1e4cfcd25501723c67f4a7fc7a13aec0c3fa
<details><summary>CTS tests (Failed: 0/138994)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu navi3x, Srdcvk</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu navi2x, Srdcvk</summary><ul>

<li>Passed: 41177/69491 (59.3%)</li>
<li>Failed: 0/69491 (0.0%)</li>
<li>Not Supported: 28314/69491 (40.7%)</li><li>Warnings: 0/69491 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]

**Pull request merged**